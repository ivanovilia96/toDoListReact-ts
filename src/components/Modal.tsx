import React from "react";
import {Modal as ModalAntD, Button} from 'antd';
import {
    Link,
} from "react-router-dom";

export default function Modal(props: { title: string, description: string }) {
    const {title, description} = props
    return (
        <ModalAntD
            closable={false}
            title={title}
            visible={true}
            footer={
                <Link to={'/'}>
                    <Button>go to the main page</Button>
                </Link>}>
            {description}
        </ModalAntD>
    )
}