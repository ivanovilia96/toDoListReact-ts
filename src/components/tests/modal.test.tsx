import React from 'react';
import Modal from '../Modal';
import {shallow} from "enzyme";

test('renders the component Modal', () => {
    const component = shallow(<Modal title={'mockTitle'} description={'mock-description'}/>);
    expect(component).toMatchSnapshot();
});