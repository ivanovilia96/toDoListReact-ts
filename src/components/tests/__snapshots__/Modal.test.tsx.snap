// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Modal
  closable={false}
  confirmLoading={false}
  footer={
    <Link
      to="/"
    >
      <Button>
        go to the main page
      </Button>
    </Link>
  }
  maskTransitionName="fade"
  okType="primary"
  title="mockTitle"
  transitionName="zoom"
  visible={true}
  width={520}
>
  mock-description
</Modal>
`;
